SPECS/zstd/zstd.spec

```diff
@@ -1,96 +1,323 @@
-Requires:       %{name} = %{version}-%{release}
+# For backwards compatibility
+Provides:       zstd-devel = %{version}-%{release}
+Obsoletes:      zstd-devel <= 1.5.5-5
 
-%description    doc
-Documentation files for zstd
+%package -n lib%{name}-static
+Summary:        Static variant of the Zstd library
+Requires:       lib%{name}-devel = %{version}-%{release}
+
+%description -n lib%{name}-devel
+Header files for Zstd library.
+
+%description -n lib%{name}-static
+Static variant of the Zstd library.
 
 %prep
-%autosetup
+%setup -q
+find -name .gitignore -delete
+%if %{with pzstd}
+%patch1 -p1
+%endif
 
 %build
-%make_build
+export CFLAGS="$RPM_OPT_FLAGS"
+export LDFLAGS="$RPM_LD_FLAGS"
+export PREFIX="%{_prefix}"
+export LIBDIR="%{_libdir}"
+%make_build -C lib lib-mt %{!?with_asm:ZSTD_NO_ASM=1}
+%make_build -C programs %{!?with_asm:ZSTD_NO_ASM=1}
+%if %{with pzstd}
+export CXXFLAGS="$RPM_OPT_FLAGS"
 %make_build -C contrib/pzstd %{!?with_asm:ZSTD_NO_ASM=1}
+%endif
+
+%check
+execstack lib/libzstd.so.1
+
+export CFLAGS="$RPM_OPT_FLAGS"
+export LDFLAGS="$RPM_LD_FLAGS"
+make -C tests test-zstd
+%if %{with pzstd} && %{with gtest}
+export CXXFLAGS="$RPM_OPT_FLAGS"
+make -C contrib/pzstd test
+%endif
 
 %install
-%make_install prefix=%{_prefix}
-find %{buildroot} -type f -name "*.a" -delete -print
+%make_install PREFIX=%{_prefix} LIBDIR=%{_libdir}
+%if %{with pzstd}
 install -D -m755 contrib/pzstd/pzstd %{buildroot}%{_bindir}/pzstd
+install -D -m644 programs/%{name}.1 %{buildroot}%{_mandir}/man1/p%{name}.1
+%endif
 
-%check
-%make_build check
+%files
+%doc CHANGELOG README.md
+%{_bindir}/%{name}
+%if %{with pzstd}
+%{_bindir}/p%{name}
+%{_mandir}/man1/p%{name}.1*
+%endif
+%{_bindir}/%{name}mt
+%{_bindir}/un%{name}
+%{_bindir}/%{name}cat
+%{_bindir}/%{name}grep
+%{_bindir}/%{name}less
+%{_mandir}/man1/%{name}.1*
+%{_mandir}/man1/un%{name}.1*
+%{_mandir}/man1/%{name}cat.1*
+%{_mandir}/man1/%{name}grep.1*
+%{_mandir}/man1/%{name}less.1*
+%license COPYING LICENSE
 
-%ldconfig_scriptlets
+%files -n lib%{name}
+%{_libdir}/libzstd.so.*
+%license COPYING LICENSE
 
-%files
-%defattr(-,root,root)
-%{_bindir}/unzstd
-%{_bindir}/zstd
-%{_bindir}/pzstd
-%{_bindir}/zstdless
-%{_bindir}/zstdmt
-%{_bindir}/zstdgrep
-%{_bindir}/zstdcat
-
-%files devel
-%{_includedir}/*.h
+%files -n lib%{name}-devel
+%{_includedir}/zdict.h
+%{_includedir}/zstd.h
+%{_includedir}/zstd_errors.h
 %{_libdir}/pkgconfig/libzstd.pc
 %{_libdir}/libzstd.so
 
-%files libs
-%license LICENSE COPYING
-%{_libdir}/libzstd.so.1*
+%files -n lib%{name}-static
+%{_libdir}/libzstd.a
 
-%files doc
-%{_mandir}/man1/*
+%ldconfig_scriptlets -n lib%{name}
 
 %changelog
-* Tue Jan 16 2024 Andrew Phelps <anphel@microsoft.com> - 1.5.5-2
-- Add pzstd binary
+* Wed Jan 31 14:14:49 EST 2024 Dan Streetman <ddstreet@ieee.org> - 1.5.5-5
+- Update to version from Fedora 39.
+- Next line is present only to avoid tooling failures, and does not indicate the actual package license.
```

Comment:
Just a note; I believe the (MIT) license on the next line is for the Fedora SPEC, not for the zstd source

Reply:
side note; we might need to update the following to use `Initial Azure Linux import` https://github.com/microsoft/CBL-Mariner/blob/3.0-dev/toolkit/scripts/spec_source_attributions.py

Reply:
I'm not sure why our toolkit thinks all fedora spec files are licensed MIT, as that is definitely not correct; but it does *require* the line to match exactly that, so I just noted that the MIT license isn't necessarily the license actually in use (either for the spec file or the package).